File: problems/11_Container_With_Most_Water.cpp
#include <bits/stdc++.h>

using namespace std;

int maxArea(vector<int>& height) {
  int water = INT_MIN;
  int i = 0, j = height.size()-1;
  while(i<j) {
    water = max(water, (j-i)*min(height[i],height[j]));
    if(height[i]<height[j]) {
      i++;
    } else {
      j--;
    }
  }
  return water;
}

int main() {
  vector<int> heights;
  heights.push_back(1);
  heights.push_back(2);
  heights.push_back(8);
  heights.push_back(7);
  cout << maxArea(heights);
  
}

File: problems/11_Container_With_Most_Water.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

def maxArea(height):
    maxInt = 0
    for i in range(0,len(height)//2):
        for j in range(i+1,len(height)):
            area = abs(i-j)*min(height[i],height[j]);
            if area > maxInt:
                maxInt = area
    return maxInt

containers = [1,8,6,2,5,4,8,3,7]
print(maxArea(containers))
